kratos/includes/checks.h

```diff
@@ -44,6 +44,8 @@ KRATOS_ERROR << "The string \"" << SubString << "\" was not found in the given s
 
 #define KRATOS_CHECK_NEAR(a,b, tolerance) if(std::abs(a - b) > tolerance) KRATOS_ERROR << "Check failed because " << #a << " = " << a << \
 " is not near to " << #b << " = " << b << " within the tolerance " << tolerance
+#define KRATOS_CHECK_RELATIVE_NEAR(a,b, tolerance) if(std::abs(b) > std::numeric_limits<double>::epsilon()) { KRATOS_ERROR_IF(std::abs((a - b)/b) > tolerance) << "Check failed because " << #a << " = " << a << \
```

Comment:
I don't understand why the `if` here is needed, can you briefly explain pls?

Reply:
Because the value of the reference can be near to zero, then division by zero, so instead of a relative comparison an absolute one is performed

Reply:
yes that makes sense ... :man_facepalming: